.github/workflows/test-workflow.yml

```diff
@@ -0,0 +1,47 @@
+name: Test workflow
+
+on:
+  workflow_call:
+    inputs:
+      node-version:
+        required: true
+        type: number
+
+jobs:
+  test:
+    name: '${{ matrix.package }} - node.js v${{ inputs.node-version }}'
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        package:
+          - apollo-server-graphql-api
+          - async-setup-with-delay
+          - async-setup-with-file
+          - babel
+          - express-rest-api
+          - karma
+          - node-sqlite3
+          - playwright
+          - programmatic-usage
+          - puppeteer
+          - react-webpack
+          - rxjs
+          - selenium
+          - third-party-reporter
+          - typescript
+          - typescript-babel
+          - vue-puppeteer
```

Comment:
[Refactor] This duplicates the information on disk, that `packages/*` contains all packages. It'd be easy for someone to make a change in one place (e.g. adding an example package) and not in the other. Could you source this from an `ls -l` call instead please? 

https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs#example-using-an-output-to-define-two-matrices

I suppose either this workflow or the main `test.yml` could do that call... My gut is that passing it as a workflow call input would be cleaner - so that each package doesn't incur a file system read. But that's not a huge difference.